crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -140,10 +143,182 @@ impl Persistence {
 
         ex.commit().await.context("commit")
     }
+
+    /// Returns an atomic transaction object which can be used to guarantee
+    /// multiple persistence operation happen consistently.
+    pub async fn begin(&self) -> Result<transaction::Transaction, Error> {
+        transaction::Transaction::begin(&self.postgres).await
+    }
+
+    /// Returns the latest block number for which settlement events have been
+    /// saved.
+    pub async fn latest_settlement_event_block(&self) -> Result<u64, Error> {
```

Comment:
This name is not accurate since we could also return a block where only an invalidation or pre-signature happened.

Reply:
The goal is to return the latest event that was indexed for the settlement contract (but I see how the name is confusing). 
In another PR I tried to combine all our event indexer to just have a single `latest_event_block` message but that PR turned out to be a huge mess.